src/menu/rendersettingsmenustate.cpp

```diff
@@ -352,16 +352,28 @@ RenderSettingsMenuState::RenderSettingsMenuState(const std::shared_ptr<MenuRingT
     {
       toggle(engine, engine.getEngineConfig()->renderSettings.fullscreen);
     });
+ 
+   m_dustDensitySelector = std::make_shared<ui::widgets::ValueSelector<float>>(
+    [](float value)
+    {
+      return /* translators: TR charmap encoding */ _("\x1f\x6c %1% \x1f\x6d Dust Particles",
+                                                      static_cast<float>(value));
+    },
+    std::vector<float>{100.0f, 50.0f, 33.33f, 25.0, 20.0f, 16.66f, 14.28f, 12.5f});
   listBox->addSetting(
-    /* translators: TR charmap encoding */ _("Dust"),
+    gslu::nn_shared<ui::widgets::Widget>{m_dustDensitySelector},
     [&engine]()
     {
       return engine.getEngineConfig()->renderSettings.dust;
     },
     [&engine]()
     {
-      toggle(engine, engine.getEngineConfig()->renderSettings.dust);
+       toggle(engine, engine.getEngineConfig()->renderSettings.dust);
     });
+  while(m_dustDensitySelector->getSelectedValue()
+        != engine.getEngineConfig()->renderSettings.dustDensity)
+        m_dustDensitySelector->selectNext();
```

Comment:
This is a bad approach I chose when implementing the other occurences of this pattern, because it will lead to an endless loop when trying to pre-select a value that's not in the known values anymore. A more clear approach would be to add a function like this to the `ValueSelector` (don't assume it will work out of the box, this is from memory):

```cpp
void selectValue(const T& value)
{
  if(const auto it = m_values.find(value); it != m_values.end())
    m_selectedValue = std::distance(m_values.begin(), it);
}
```